server/registry.go

```diff
@@ -166,81 +169,127 @@ func newHandlerRegistry(name string, logger *zap.SugaredLogger, requestValidator
+			hDTO.Path = np
+		}
+	}
 
-				if h.Config().Consumes != "" {
-					hDTO.Consumes = h.Config().Consumes
-				} else {
-					hDTO.Consumes = "application/json"
-				}
+	// Prepend the controller validator if defined, so that the controller validator is called first.
+	if c, ok := controller.(IControllerAuthZValidator); ok {
+		var simpleHandler AuthZValidatorV2Fn
 
-				mt, err := contenttype.ParseMediaType(hDTO.Produces)
-				if err != nil {
-					return nil, multierr.Append(
-						fmt.Errorf("failed to process mime type (%s) for handler with method: %s, path: %s", hDTO.Produces, hDTO.Method, hDTO.Path),
-						err,
-					)
-				}
-				hDTO.MediaType = mt
+		simpleHandler = func(ctx context.Context, p *iam.ArmoryCloudPrincipal) (string, bool) {
+			return c.AuthZValidator(p)
+		}
+		validators = append(validators, simpleHandler)
+	}
+	if c, ok := controller.(IControllerAuthZValidatorV2); ok {
+		validators = append(validators, c.AuthZValidator)
+	}
 
-				if hDTO.StatusCode == 0 {
-					hDTO.StatusCode = http.StatusOK
-				}
+	var iResponseProcessors []ResponseProcessorWithOrder
+	if c, ok := controller.(IControllerPostResponseProcessor); ok {
+		iResponseProcessors = c.ResponseProcessors()
+	}
+	sort.Slice(iResponseProcessors, func(i, j int) bool {
+		return iResponseProcessors[i].Order < iResponseProcessors[j].Order
+	})
 
-				key := handlerDTOKey{
-					path:   hDTO.Path,
-					method: hDTO.Method,
-				}
+	processors := lo.Map(iResponseProcessors, func(processor ResponseProcessorWithOrder, _ int) ResponseProcessorFn {
+		return processor.Processor
+	})
 
-				if registryData[key] == nil {
-					registryData[key] = make(map[string]*handlerDTO)
-				}
+	perHandlerProcessors := handler.Config().responseProcessors
+	if len(perHandlerProcessors) > 0 {
+		processors = append(processors, perHandlerProcessors...)
+	}
 
-				if registryData[key][hDTO.Produces] != nil {
-					return nil, fmt.Errorf("failed to register hander for [Path: %s, Method: %s, Produces: %s] there was already a registered handler", hDTO.Path, hDTO.Method, hDTO.Produces)
-				}
+	hDTO.ResponseProcessors = processors
 
-				hDTO.HandlerFn = h.GetGinHandlerFn(logger, requestValidator, hDTO)
-				registryData[key][hDTO.Produces] = hDTO
-			}
-		}
+	if handler.Config().Produces != "" {
+		hDTO.Produces = handler.Config().Produces
+	} else {
+		hDTO.Produces = "application/json"
 	}
 
-	return &handlerRegistry{
-		name:   name,
-		logger: logger,
-		data:   registryData,
-	}, nil
+	if handler.Config().Consumes != "" {
+		hDTO.Consumes = handler.Config().Consumes
+	} else {
+		hDTO.Consumes = "application/json"
+	}
+
+	mt, err := contenttype.ParseMediaType(hDTO.Produces)
+	if err != nil {
+		return multierr.Append(
+			fmt.Errorf("failed to process mime type (%s) for handler with method: %s, path: %s", hDTO.Produces, hDTO.Method, hDTO.Path),
+			err,
+		)
+	}
+	hDTO.MediaType = mt
+
+	if hDTO.StatusCode == 0 {
```

Comment:
Status code "0" usually means timeout or something. Is it safe to return a "200"? Are we hiding errors when we do this?